### PR TITLE
Add the problem statement tab

### PR DESCRIPTION
**Pull request by Aritra8438**
This PR adds the `problem statement` tab to the creator page..

Fixes: #7587

https://github.com/omegaup/omegaup/assets/64671908/049bccc2-f6d9-41b6-b801-961cacec625f

# Comments:
Following the unofficial CDP, we will update the markdown content and the store only on the `Save/Trigger` button. The content will be stored in the store which will be used to generate the zip file.

This PR
- [x] adds the problem statement tab to the creator.
- [ ] Writes the `jest` test for it.

# Checklist:

- [x] The code follows the [coding guidelines](https://github.com/omegaup/omegaup/wiki/Coding-guidelines) of omegaUp.
- [x] The tests were executed and all of them passed.
- [x] If you are creating a feature, the new tests were added.
- [x] If the change is large (> 200 lines), this PR was split into various Pull Requests. It's preferred to create one PR for changes in controllers + unit tests in PHPUnit,  and then another Pull Request for UI + tests in Jest, Cypress or both.

### CODE REVIEW

**Comment by Aritra8438** [comment]
@carlosabcs, PTAL!

I will soon add the tests to the PR.

**Review by carlosabcs — COMMENTED** [review]
The PR looks nice, just consider the comments I've added and include the tests.

File: frontend/www/js/omegaup/components/problem/creator/Tabs.vue [hunk 1]
```diff
@@ -31,9 +32,10 @@
 <script lang="ts">
 import { Component, Vue } from 'vue-property-decorator';
 import cases_Tab from './cases/CasesTab.vue';
+import statement_Tab from './statement/StatementTab.vue';
 import T from '../../../lang';
 @Component({
-  components: { 'cases-tab': cases_Tab },
+  components: { 'statement-tab': statement_Tab, 'cases-tab': cases_Tab },
```
[the remark this answers is not in the document]
Comment (Aritra8438) [reply frontend/www/js/omegaup/components/problem/creator/Tabs.vue:38]: Okay @carlosabcs, but what about the cases-tab component that's already implemented? Should I change it in this PR or dedicated PR?

File: frontend/www/js/omegaup/components/problem/creator/statement/StatementTab.vue [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      panels: {
+        buttonBar: this.markdownButtonBar,
+        preview: null,
+        input: this.markdownInput,
+      },
+    });
+    this.markdownEditor.run();
+  }
+
+  triggerMarkdown() {
```
[the remark this answers is not in the document]
Comment (Aritra8438) [reply frontend/www/js/omegaup/components/problem/creator/statement/StatementTab.vue:73]: Okayy.

File: frontend/templates/en.lang [hunk 3]
```diff
@@ -1038,6 +1039,7 @@ problemCreatorSolution = "Solution"
 problemCreatorStatement = "Statement"
 problemCreatorSuffix = "Suffix"
 problemCreatorTestCases = "Test cases"
+problemCreatorTriggerMarkdown = "Save"
```
[the remark this answers is not in the document]
Comment (Aritra8438) [reply frontend/templates/en.lang:1042]: As we are saving markdown only, should I rename it to `problemCreatorMarkdownSave`?

**Comment by Aritra8438** [comment]
I've fixed it @carlosabcs. PTAL!

**Comment by heduenas** [comment]
@Aritra8438 when Carlos approves your PRs please request my review through github.

File: frontend/templates/en.lang (hunk 3 above)
[the remark this answers is not in the document]
Comment (carlosabcs) [reply frontend/templates/en.lang]: Sure!

File: frontend/www/js/omegaup/components/problem/creator/Tabs.vue [hunk 4]
```diff
@@ -31,9 +32,13 @@
 <script lang="ts">
 import { Component, Vue } from 'vue-property-decorator';
 import cases_Tab from './cases/CasesTab.vue';
+import problemCreator_Statement_Tab from './statement/StatementTab.vue';
```
Comment (carlosabcs) [inline frontend/www/js/omegaup/components/problem/creator/Tabs.vue:35]:
```suggestion
import problemCreator_StatementTab from './statement/StatementTab.vue';
```

File: frontend/www/js/omegaup/components/problem/creator/Tabs.vue [hunk 5]
```diff
@@ -31,9 +32,13 @@
...
 import T from '../../../lang';
 @Component({
-  components: { 'cases-tab': cases_Tab },
+  components: {
+    'omegaup-statement-tab': problemCreator_Statement_Tab,
```
Comment (carlosabcs) [inline frontend/www/js/omegaup/components/problem/creator/Tabs.vue:39]:
```suggestion
    'omegaup-problem-creator-statement-tab': problemCreator_StatementTab,
```

File: frontend/www/js/omegaup/components/problem/creator/Tabs.vue [hunk 6]
```diff
@@ -31,9 +32,13 @@
 <script lang="ts">
 import { Component, Vue } from 'vue-property-decorator';
 import cases_Tab from './cases/CasesTab.vue';
```
Comment (carlosabcs) [inline frontend/www/js/omegaup/components/problem/creator/Tabs.vue:34]: Maybe create another issue to rename this to `problemCreator_CasesTab`.

Comment (carlosabcs) [reply frontend/www/js/omegaup/components/problem/creator/Tabs.vue:34]: What do you think @heduenas ?

**Review by carlosabcs — APPROVED** [review]
LGTM. Just small corrections.

**Comment by Aritra8438** [comment]
@heduenas, PTAL!

**Review by heduenas — APPROVED** [review]

File: frontend/www/js/omegaup/components/problem/creator/Tabs.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (Aritra8438) [reply frontend/www/js/omegaup/components/problem/creator/Tabs.vue]: This is tracked in the following issue:
- [ ] #7648

**Pull request merged**